server/log.go

```diff
@@ -37,13 +37,17 @@ type Logger interface {
 
 // SetLogger sets the logger of the server
 func (s *Server) SetLogger(logger Logger, debugFlag, traceFlag bool) {
+	debugVal := 0
 	if debugFlag {
-		atomic.StoreInt32(&debug, 1)
+		debugVal = 1
```

Comment:
Allowing the call to set debug/trace to 0, not only 1. It used that you would only set it to 1, but never be able to set it back to 0.